Tests/NIOCoreTests/AsyncChannel/AsynChannelUnixDomainSocketTests.swift

```diff
@@ -0,0 +1,171 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2025 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+#if canImport(Testing)
+import NIOPosix
+import Testing
+
+@testable import NIOCore
+
+@Suite
+private enum AsynChannelUnixDomainSocketTests {
+    /// This is a end-to-end async channel based test.
+    ///
+    /// The server side listens on a UNIX domain socket, and the client connects to this socket.
+    ///
+    /// The server and client exchange simple, line based messages.
+    @available(macOS 14.0, iOS 17, tvOS 17, watchOS 10, *)
+    @Test()
+    static func runServer() async throws {
+        try await confirmation("Client did receive message") { clientDidReceive in
+            try await confirmation("Server did receive message") { serverDidReceive in
+                try await check(
+                    clientDidReceive: clientDidReceive,
+                    serverDidReceive: serverDidReceive
+                )
+            }
+        }
+    }
+}
+
+@available(macOS 14.0, iOS 17, tvOS 17, watchOS 10, *)
+private func check(
+    clientDidReceive: Confirmation,
+    serverDidReceive: Confirmation
+) async throws {
+    // This uses a hard-coded path.
+    //
+    // The path of a UNIX domain socket has a relatively low limit on its total
+    // length, and we thus can not put this inside some (potentially) deeply
+    // nested directory hierarchy.
+    let path = "/tmp/9ac7750dc22a066066871aadf481e31a"
+    let serverChannel = try await makeServerChannel(path: path)
+
+    try await withThrowingDiscardingTaskGroup { group in
+        try await serverChannel.executeThenClose { inbound in
+            group.addTask {
+                // Create a client connection to the server:
+                let clientChannel = try await makeClientChannel(path: path)
+                print("Executing client channel")
```

Comment:
Can you remove the prints?

Reply:
Even in test code? This is super helpful for diagnosing these when they fail.